cmd/kops/create.go

```diff
@@ -192,13 +192,13 @@ func RunCreate(ctx context.Context, f *util.Factory, out io.Writer, c *CreateOpt
 					return err
 				}
 
-				sshCredentialStore, err := clientset.SSHCredentialStore(cluster)
+				sshCredentialStore, err := clientset.SSHCredentialStore(ctx, cluster)
 				if err != nil {
 					return err
 				}
 
 				sshKeyArr := []byte(v.Spec.PublicKey)
-				err = sshCredentialStore.AddSSHPublicKey(sshKeyArr)
+				err = sshCredentialStore.AddSSHPublicKey(ctx, sshKeyArr)
```

Comment:
So you have to pass a context to create a `SSHCredentialStore` and then you have to pass it again to call a receiver on that `SSHCredentialStore`. This just doesn't make sense to me.

I think you'd be better off passing a `VFSContext` to create the `simple.Clientset`. Then you'd be able to have tests create a mock `VFSContext`.

Reply:
I think the core of it is that all methods are going to take a context.

Once all methods take a context, then we can make good choices regarding where to consume values from the context.  It's hard right now to start using the context, because it's tempting to use the subset of methods that have a context.  (As you've seen, where I was trying to add context partially and use it in the wrong places).

Reply:
One doesn't pass a context to `k8sClient.CoreV1()` or `k8sClient.CoreV1().Nodes()`. Similarly, one shouldn't pass a context to `clientset.SSHCredentialStore()`.

Reply:
I'm proposing we just add them, then we can prune if we feel they aren't needed.  Otherwise this is going to take a very long time....